contracts/test/mocks/MockGMPGatewayWithToken.sol

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IAxelarGMPGatewayWithToken } from '../../interfaces/IAxelarGMPGatewayWithToken.sol';
+import { IERC20 } from '../../interfaces/IERC20.sol';
+import { IERC20MintableBurnable } from '../../interfaces/IERC20MintableBurnable.sol';
+
+import { ERC20MintableBurnable } from '../token/ERC20MintableBurnable.sol';
+
+contract MockGMPGatewayWithToken is IAxelarGMPGatewayWithToken {
```

Comment:
this is not needed atm. We already have `MockGateway` which has both GMP and token functionality. We can split that into MockGMPGateway and MockGMPGatewayWithToken in a separate PR

Reply:
got it. updated.